src/VirtualContent.js

```diff
@@ -0,0 +1,341 @@
+function composedTreeParent(node) {
+  return node.assignedSlot || node.host || node.parentNode;
+}
+
+function nearestScrollingAncestor(node) {
+  for (node = composedTreeParent(node); node !== null; node = composedTreeParent(node)) {
+    if (node.nodeType === Node.ELEMENT_NODE && node.scrollHeight > node.clientHeight) {
+      return node;
+    }
+  }
+  return null;
+}
+
+const DEFAULT_HEIGHT_ESTIMATE = 100;
+const TEMPLATE = `
+<style>
+:host {
+  /* Use flex to prevent children's margins from collapsing. Avoiding margin
+   * collapsing is simpler and good enough to start with. */
+  display: flex;
+  flex-direction: column;
+
+  /* Prevent the automatic scrolling between writes and later measurements,
```

Comment:
"writes" here just refers to DOM modifications. Basically, without `overflow-anchor: none`, I noticed that the browser would automatically change the scroll position after some DOM edits. However, I'm trying to manually make those adjustments myself in `_update`, so I needed to prevent this behavior so that they wouldn't conflict.